nlg/search.py

```diff
@@ -565,6 +568,7 @@ def templatize(text, args, df):
     The iris dataset has 3 {{ df.columns[0] }} - {{ df["species"].iloc[0] }}, \
 {{ df["species"].iloc[1] }} and {{ df["species"].iloc[-1] }}.
     """
+    text = nlp(numerize(str(text)))
```

Comment:
Accept two kinds of input: a spacy Doc or a Python string. If the input is a spacy doc:
1. numerize it as `numerize(text.text)` and remap it back.
2. If it's text, convert to a spacy doc first.